From 2d9ddcc8ac93b31a404837c19cce8971b80ba032 Mon Sep 17 00:00:00 2001
From: Ludovic Barre <ludovic.barre@foss.st.com>
Date: Mon, 23 Sep 2024 15:37:34 +0200
Subject: [PATCH 1275/1695] product: tfm-stm32mp2: move std=c11 compiler flag

To override the standard on c11 (needed for scp), the std=c11 option
must be the last "std" argument of C_FLAGS list.
CMakeLists.txt adds std option before c99, so move add_compile_options
to Buildoptions.cmake.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git c74766945be4a0ab7bad9483e6016607148e5c52 product: tfm-stm32mp2: move std=c11 compiler flag]
Change-Id: I30ebce5d95c66052583af3a210123bf7c0c312b5
Signed-off-by: Ludovic Barre <ludovic.barre@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/SCP-firmware/+/406166
Domain-Review: Etienne CARRIERE <etienne.carriere@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Etienne CARRIERE <etienne.carriere@foss.st.com>
---
 .../product/tfm-stm32mp2/fw/Buildoptions.cmake        | 11 +++++++++++
 .../product/tfm-stm32mp2/fw/CMakeLists.txt            |  5 +----
 2 files changed, 12 insertions(+), 4 deletions(-)
 create mode 100644 core/lib/scmi-server/SCP-firmware/product/tfm-stm32mp2/fw/Buildoptions.cmake

diff --git a/core/lib/scmi-server/SCP-firmware/product/tfm-stm32mp2/fw/Buildoptions.cmake b/core/lib/scmi-server/SCP-firmware/product/tfm-stm32mp2/fw/Buildoptions.cmake
new file mode 100644
index 000000000..3ec440500
--- /dev/null
+++ b/core/lib/scmi-server/SCP-firmware/product/tfm-stm32mp2/fw/Buildoptions.cmake
@@ -0,0 +1,11 @@
+#
+# Arm SCP/MCP Software
+# Copyright (c) 2024, STMicroelectronics and the Contributors. All rights reserved.
+#
+# SPDX-License-Identifier: BSD-3-Clause
+#
+
+#
+# Ensure TF-M build-system does not override requested C standard.
+#
+add_compile_options(-std=c11)
diff --git a/core/lib/scmi-server/SCP-firmware/product/tfm-stm32mp2/fw/CMakeLists.txt b/core/lib/scmi-server/SCP-firmware/product/tfm-stm32mp2/fw/CMakeLists.txt
index 341869942..e0b203445 100644
--- a/core/lib/scmi-server/SCP-firmware/product/tfm-stm32mp2/fw/CMakeLists.txt
+++ b/core/lib/scmi-server/SCP-firmware/product/tfm-stm32mp2/fw/CMakeLists.txt
@@ -28,8 +28,5 @@ target_sources(
     PRIVATE "${CMAKE_CURRENT_SOURCE_DIR}/config_all.c"
 )
 
-#
-# Ensure TF-M build-system does not override requested C standard.
-#
 
-add_compile_options(-std=gnu11)
+
-- 
2.47.2

